### PR TITLE
Add all_frames:true to browser devtool extension

### PR DESCRIPTION
**Pull request by dmail**
## Summary

This pull requests aims to improve iframe support for devtools browser extensions.

List of changes regarding this pull request in chronological order
---
1. Open Issue to Improve browser extension iframe support (https://github.com/facebook/react/issues/18945).
2. Push commits adding `"all_frames": true` to browser extensions, write test plan, move pull request to `Ready for review`.
3. @bvaughn request changes because Component and Profiler tabs are not working.
4. **Need help** to fix Component and Profiler tabs.

### Before

chromeko

### After

chromeok

## Setup

We want to see how react devtools works in the following iframe scenarios:

- An iframe on an other origin
   ```html
   <iframe src="http://127.0.0.1:3000/main.html"></iframe>
   ```
  
   This scenario requires to run the http-server running on port 3000.

   ```console
   node fixtures/devtools/iframe/server.js
  ```
  Then open [fixtures/devtools/iframe/iframe-other-origin.html](https://github.com/facebook/react/blob/94611ebe5cbe4f9d4f5adf6a9a5044aae980978f/fixtures/devtools/iframe/iframe-other-origin.html) file directly in the browser.

- An iframe on the same origin
   ```html
   <iframe src="./main.html"></iframe>
   ```

  Open [fixtures/devtools/iframe/iframe-same-origin.html](https://github.com/facebook/react/blob/94611ebe5cbe4f9d4f5adf6a9a5044aae980978f/fixtures/devtools/iframe/iframe-same-origin.html) file directly in the browser.

- An iframe on the same origin, sandboxed but allowed to run scripts
   ```html
   <iframe sandbox="allow-scripts" src="./main.html"></iframe>
   ```

   Open [fixtures/devtools/iframe/iframe-same-origin-sandbox.html](https://github.com/facebook/react/blob/94611ebe5cbe4f9d4f5adf6a9a5044aae980978f/fixtures/devtools/iframe/iframe-same-origin-sandbox.html) file directly in the browser.

In each scenario browser devtools must have fully functionnal ⚛️ Components and ⚛️ Profiler panels.

### Chrome extension

1. cd packages/react-devtools-extensions/
2. yarn build:chrome
3. yarn run test:chrome
4. Open iframe-other-origin.html in chrome as described in the test plan setup
5. Open iframe-same-origin.html in chrome as described in the test plan setup
6. Open iframe-same-origin-sandbox.html in chrome as described in the test plan setup

For 4, 5, 6 chrome extension must be able to detect react that is inside the iframe.

chromeok

### Edge extension

1. cd packages/react-devtools-extensions/
2. yarn build:edge
3. yarn run test:edge
4. Open iframe-other-origin.html in chrome as described in the test plan setup
5. Open iframe-same-origin.html in chrome as described in the test plan setup
6. Open iframe-same-origin-sandbox.html in chrome as described in the test plan setup

For 4, 5, 6 edge extension must be able to detect react that is inside the iframe.

edgeok

### Firefox extension

1. cd packages/react-devtools-extensions/
2. yarn build:firefox
3. yarn run test:firefox
At this step it's possible that one of this happens:
(a) devtool extension fails to detect react even on reactjs.org 
(b) devtool extension does not show up in firefox toolbar
Both of these behaviour are already on `master` and were not introduced in this pull request.
If that happens you can workaround the buggy situation.
(a) Refresh the page
(b) Go to firefox settings and toggle react devtool addon
4. Open iframe-other-origin.html in chrome as described in the test plan setup
5. Open iframe-same-origin.html in chrome as described in the test plan setup
6. Open iframe-same-origin-sandbox.html in chrome as described in the test plan setup

For 4, 5 firefox extension must be able to detect react that is inside the iframe.

firefoxok

For 6, Firefox is not executing the extension in a sandboxed iframe so react cannot be detected.

### CODE REVIEW

File: packages/react-devtools-extensions/iframe/iframe-other-domain.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE html>
+<html>
+  <head></head>
+  <body>
+    <iframe src="http://127.0.0.1:8080/packages/react-devtools-extensions/iframe/main.html" />
```
Comment (bvaughn) [inline packages/react-devtools-extensions/iframe/iframe-other-domain.html:5]: These files don't belong in this package. Packages are for source code. There's a separate folder for fixtures:

https://github.com/facebook/react/tree/master/fixtures/devtools

**Comment by dmail** [comment]
I was focused on making browser extension detect react. Once it was working I assumed it was enough. 
I will move iframe files in the fixtures directory and try to make ⚛️ Component and ⚛️ Profiler tabs working

File: fixtures/devtools/iframe/server.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const finalhandler = require('finalhandler');
```
Comment (bvaughn) [inline fixtures/devtools/iframe/server.js:1]: nit would be nice to make this file executable:
```sh
chmod +x ./fixtures/devtools/iframe/server.js
```

Then add a header:

```suggestion
#!/usr/bin/env node

const finalhandler = require('finalhandler');
```

Comment (dmail) [reply fixtures/devtools/iframe/server.js:3]: Done :)

**Comment by dmail** [comment]
I haven't worked on it since my comment from 19 May https://github.com/facebook/react/pull/18952#issuecomment-630916183.
When I left it, DevTools where not seeing components inside the iframe so ⚛️ Component and ⚛️ Profiler tabs where not working. 

Right now this PR adds `all_frames: true` to browser extensions `manifest.json` files and some fixtures to test iframe integration in various scenarios.
What remains to be done is to understand why ⚛️ Component and ⚛️ Profiler tabs are not working and fix them.

**Comment by bvaughn** [comment]
I think that making the extension icon work without making the Components and Profiler tabs work probably isn't all that value (and maybe confuse people further). If you decide to continue work on this, ping me when it's ready for another look.

**Comment by dmail** [comment]
Totally agree. 
The only change I made into important files is to do `all_frame: true`. I was hoping it would be sufficient for the extension to work in iframes. When I saw the icon working properly I assumed it was fine and asked for a review. In the end the icon is working but not the tabs and it is confusing.

As I said I am kinda blocked and I am focused on other things at the moment.
I will close this PR to underline that it's no longer active.
If anyone wants to resume the work feel free to cherry pick commit or files :)

**Comment by bvaughn** [comment]
Fair enough, and thanks for the update!

**Pull request closed without merging**